src/pyramid_retry/__init__.py

```diff
@@ -110,7 +107,8 @@ def retry_policy(environ, router):
             if number > 0:
                 # try to make sure this code stays in sync with pyramid's
                 # router which normally creates requests
-                request = router.make_request(environ)
+                request_ctx = router.request_context(environ)
+                request = request_ctx.begin()
```

Comment:
Wouldn't this leave the threadlocal's pushed from line 76? You never seem to call `.end()` on that original `request_ctx` if number is greater than 0.